src/Mod/TechDraw/Gui/TaskDimension.h

```diff
@@ -0,0 +1,105 @@
+/***************************************************************************
+ *   Copyright (c) 2021 Uwe Stöhr <uwestoehr@lyx.org>                      *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef GUI_TASKVIEW_TASKDIMENSION_H
+#define GUI_TASKVIEW_TASKDIMENSION_H
+
+#include <Gui/TaskView/TaskView.h>
+#include <Gui/TaskView/TaskDialog.h>
+
+#include <Mod/TechDraw/Gui/ui_TaskDimension.h>
+
+#include "QGIViewDimension.h"
+#include "ViewProviderDimension.h"
+
+class Ui_TaskDimension;
+
+namespace TechDrawGui
+{
+
+class TaskDimension : public QWidget
+{
+    Q_OBJECT
+
+public:
+    TaskDimension(QGIViewDimension *parent, ViewProviderDimension *dimensionVP);
+    ~TaskDimension();
+
+public:
+    virtual bool accept();
+    virtual bool reject();
+    void recomputeFeature();
+
+private Q_SLOTS:
+    void onTheoreticallyExactChanged();
+    void onEqualToleranceChanged();
+    void onOvertoleranceChanged();
+    void onUndertoleranceChanged();
+    void onFormatSpecifierChanged();
+    void onArbitraryChanged();
+    void onToleranceFormatSpecifierChanged();
+    void onArbitraryTolerancesChanged();
+    void onFlipArrowheadsChanged();
+    void onColorChanged();
+    void onFontsizeChanged();
+    void onDrawingStyleChanged();
+
+private:
+    Ui_TaskDimension *ui;
```

Comment:
As a matter of principal I side with the C++ Core Guidelines here, I think this should be a unique_ptr to clarify that it owns the memory (and to ensure it's deleted, though of course in this case it's trivial). https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#S-resource

Reply:
> As a matter of principal I side with the C++ Core Guidelines here, I think this should be a unique_ptr

Good point. I did this now. When this change is accepted I will make a further PR since at the moment only the TD preferences dialogs use a unique_ptr.

Reply:
> When this change is accepted I will make a further PR since at the moment only the TD preferences dialogs use a unique_ptr.

I don't know that it's necessary to go back through too much old code to make this change, I am mostly concerned with trying to make sure that new code follows the guidelines, more or less. But don't let me stop you!

Reply:
>> As a matter of principal I side with the C++ Core Guidelines here, I think this should be a unique_ptr to clarify that it owns the memory (and to ensure it's deleted, though of course in this case it's trivial). https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#S-resource
> 
> Good point. I did this now. When this change is accepted I will make a further PR since at the moment only the TD preferences dialogs use a unique_ptr.

Here is the PR: #4293